scrapinghub/client/exceptions.py

```diff
@@ -69,6 +73,8 @@ def wrapped(*args, **kwargs):
                 raise ValueTooLarge(http_error=exc)
             elif 400 <= status_code < 500:
                 raise ScrapinghubAPIError(http_error=exc)
+            elif 500 <= status_code < 600:
+                raise ServerError(http_error=exc)
```

Comment:
The problem with the wrapper is that for jl format ValueError will never be catched as method returns generator in this case. I think it'd be reasonable to raise appropriate APIError for 404 (APIError.ERR_NOT_FOUND is never used) and 5xx (new err type) in _decode_response

P.S. let's change APIError error type values to strings vs current meaningless ints, e.g. `ERR_DEFAULT = 0` -> `ERR_DEFAULT = 'err_default'`